### PR TITLE
Use mamba to install/update conda-build and other tools

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Since we are already using mambaforge based images we may as well use mamba in a few more places.

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 1]
```diff
@@ -29,11 +31,13 @@ CONDARC
...
 GET_BOA=boa
 BUILD_CMD=mambabuild
+CONDA_INSTALL_CMD=mamba
 {%- else %}
 BUILD_CMD=build
+CONDA_INSTALL_CMD=conda
 {%- endif %}
 
-conda install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
+${CONDA_INSTALL_CMD} install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
```
Comment (isuruf) [inline conda_smithy/templates/build_steps.sh.tmpl:40]: I'd rather we use `mamba` everywhere.

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 2]
```diff
@@ -21,15 +23,17 @@ bash $MINIFORGE_FILE -b -p ${MINIFORGE_HOME}
...
 BUILD_CMD=build
+CONDA_INSTALL_CMD=conda
 {%- endif %}
 
 source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
 echo -e "\n\nInstalling {{ remote_ci_setup }} and conda-build."
-conda install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
+${BUILD_CMD} install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
```
Comment (chrisburr) [inline conda_smithy/templates/run_osx_build.sh.tmpl:36]:
```suggestion
${CONDA_INSTALL_CMD} install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
```

File: conda_smithy/templates/build_steps.sh.tmpl (hunk 1 above)
Comment (jakirkham) [reply conda_smithy/templates/build_steps.sh.tmpl:40]: After running into a few issues in different feedstocks where we had to switch back to conda-build, would prefer that we keep this configurable

Comment (isuruf) [reply conda_smithy/templates/build_steps.sh.tmpl:40]: How is that relevant here? This command gives the same solve across feedstocks.

Comment (jakirkham) [reply conda_smithy/templates/build_steps.sh.tmpl:40]: Right and if some combination of packages get released that mamba can't solve, all feedstocks are stuck here. Then we have to scramble to fix it (as opposed to having another release valve to take off the pressure a bit)

Comment (mariusvniekerk) [reply conda_smithy/templates/build_steps.sh.tmpl:40]: i live keeping it configurable, granted in the near future these commands will essentially be identical

**Comment by mariusvniekerk** [comment]
We might need some other bits for mamba too since the behavior for `mamba install` is to not update things already installed unlike conda.

**Comment by beckermr** [comment]
We're burning a few minutes per build of CI time because of boa pins on our CI packages and us not using current_repodata

```
+ conda install --yes --quiet conda-forge-ci-setup=3 conda-build pip boa -c conda-forge
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.
Solving environment: ...working... failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): ...working... done
```

This PR is one way to solve this. Shall we merge @conda-forge/core?

File: conda_smithy/templates/build_steps.sh.tmpl (hunk 1 above)
Comment (beckermr) [inline conda_smithy/templates/build_steps.sh.tmpl:40]:
```suggestion
{%- if build_with_mambabuild %}
${CONDA_INSTALL_CMD} update --all --yes --quiet
{%- endif %}
${CONDA_INSTALL_CMD} install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
```

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 3]
```diff
@@ -21,15 +23,17 @@ bash $MINIFORGE_FILE -b -p ${MINIFORGE_HOME}
...
 BUILD_CMD=build
+CONDA_INSTALL_CMD=conda
 {%- endif %}
 
 source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
 echo -e "\n\nInstalling {{ remote_ci_setup }} and conda-build."
-conda install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
+${CONDA_INSTALL_CMD} install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
```
Comment (beckermr) [inline conda_smithy/templates/run_osx_build.sh.tmpl:36]:
```suggestion
{%- if build_with_mambabuild %}
${CONDA_INSTALL_CMD} update --all --yes --quiet
{%- endif %}
${CONDA_INSTALL_CMD} install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
```

**Comment by jakirkham** [comment]
> This PR is one way to solve this. Shall we merge @conda-forge/core?

It's still marked draft. Is it complete?

**Comment by beckermr** [comment]
I think the outstanding issue was updating the env. I made some suggested changes that would do that. We could go with those.

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 4]
```diff
@@ -29,11 +31,16 @@ CONDARC
...
 BUILD_CMD=mambabuild
+CONDA_INSTALL_CMD=mamba
 {%- else %}
 BUILD_CMD=build
+CONDA_INSTALL_CMD=conda
 {%- endif %}
 
-conda install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
+{%- if build_with_mambabuild %}
+${CONDA_INSTALL_CMD} update --all --yes --quiet
```
Comment (isuruf) [inline conda_smithy/templates/build_steps.sh.tmpl:41]: This is going to invalidate the pins like `python=3.9` we have added in the docker image.

**Review by isuruf — CHANGES_REQUESTED** [review]

File: conda_smithy/templates/build_steps.sh.tmpl (hunk 4 above)
Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:41]: I think that is wrong. Here is mamba on my local machine:

```
$ mamba update --all --yes

Looking for: ['libcurl', 'blinker', 'pexpect', 'gettext', 'libgcrypt', 'jsonschema', 'argon2-cffi', 'pyyaml', 'tomlkit', 'matplotlib-base', 'certifi', 'mamba', 'py-lief', 'nbformat', 'six', 'pygments', 'gitpython', 'keyring', 'liblief', 'pyzmq', 'toml', 'jq', 'qtconsole', 'notebook', 'brotli', 'iniconfig', 'ncurses', 'requests', 'zeromq', 'libmamba', 'libgpg-error', 'jupyter_console', 'cloudpickle', 'pynamodb', 'bleach', 'botocore', 'parameterized', 'send2trash', 'mccabe', 'libuv', 'make', 'rfc3986', 'pyqt-impl', 'lcms2', 'filelock', 'cycler', 'attrs', 'jupyterlab_pygments', 'pkg-config', 'testpath', 'reproc-cpp', 'reproc', 'popt', 'libglib', 'tornado', 'glib-tools', 'backcall', 'cffi', 'pandoc', 'ruamel.yaml.jinja2', 'pyparsing', 'libcxx', 'glob2', 'python-dateutil', 'zlib', 'boa', 'kiwisolver', 'pyopenssl', 'jwcrypto', 'conda-forge-pinning', 'liblapack', 'pysocks', 'coverage', 'libpq', 'uritemplate', 'lerc', 'nest-asyncio', 'more-itertools', 'simplejson', 'appdirs', 'qt', 'smmap', 'pyrsistent', 'libwebp-base', 'importlib_metadata', 'pcre', 'libedit', 'conda-build', 'ptyprocess', 'commonmark', 'pyasn1', 'python-libarchive-c', 'packaging', 'glib', 'numpy', 'jupyter_highlight_selected_word', 'prometheus_client', 'conda-package-handling', 'github3.py', 'pycosat', 'pandocfilters', 'python_abi', 'setproctitle', 'libiconv', 'ipywidgets', 'pyqtwebengine', 'nb_conda_kernels', 'pyjwt', 'munkres', 'backports', 'mysql-common', 'platformdirs', 'olefile', 'cfitsio', 'wheel', 'webencodings', 'xz', 'charset-normalizer', 'chardet', 'jinja2', 'joblib', 'git', 'brotli-bin', 'pip', 'perl', 'appnope', 'setuptools-scm', 'zipp', 'curl', 'license-expression', 'brotlipy', 'colorama', 'gitdb', 'nbclient', 'psutil', 'toolz', 'requests-toolbelt', 'rpm-tools', 'ipython', 'scrypt', 'tk', 'jbig', 'boto3', 'ruamel_yaml', 'jupyter_contrib_nbextensions', 'archspec', 'requests-mock', 'ipykernel', 'libtiff', 'click', 'libgfortran', 'tomli', 'jpeg', 'py', 'pathspec', 'pycrypto', 'sqlite', 'libbrotlidec', 'krb5', 'soupsieve', 'watchgod', 'oauthlib', 'prompt-toolkit', 'hexalattice', 'conda-suggest-conda-forge', 'readme_renderer', 'libcblas', 'widgetsnbextension', 'black', 'docutils', 'libdeflate', 'icu', 'llvm-openmp', 'libffi', 'requests-oauthlib', 'jedi', 'defusedxml', 'typing-extensions', 'uritemplate.py', 'libnghttp2', 'jupyter', 'libdb', 'prompt_toolkit', 'conda-smithy', 'rich', 'libsolv', 'jupyter_latex_envs', 'mysql-libs', 'zstd', 'mock', 'nspr', 'libopenblas', 'file', 'libarchive', 'jupyter_contrib_core', 'pyqtchart', 'frozendict', 'jupyter_nbextensions_configurator', 'tar', 'typed-ast', 'libllvm11', 'm2r', 'importlib_resources', 'yaml', 'jupyter_core', 'entrypoints', 'readline', 'openssl', 'ndg-httpsclient', 'libxml2', 'twine', 'libbrotlicommon', 'anaconda-client', 'loky', 'markupsafe', 'boolean.py', 'pluggy', 'libzlib', 'bzip2', 'libmagic', 'c-ares', 'jupyter_client', 'beautifulsoup4', 'json5', 'lazyasd', 'python', 'cmarkgfm', 'pkginfo', 'ripgrep', 'nodejs', 's3transfer', 'et_xmlfile', 'pytest-cov', 'argcomplete', 'libxslt', 'unicodedata2', 'pyqt5-sip', 'libbrotlienc', 'ruamel.yaml', 'setuptools', 'libssh2', 'wcwidth', 'yaml-cpp', 'matplotlib', 'jupyterlab_widgets', 'libsodium', 'jdcal', 'clyent', 'pygithub', 'pcre2', 'isodate', 'codecov', 'lz4-c', 'ca-certificates', 'pillow', 'python.app', 'importlib-metadata', 'idna', 'async_generator', 'pytest', 'mypy_extensions', 'tqdm', 'cachetools', 'pycodestyle', 'wurlitzer', 'qtpy', 'nss', 'jmespath', 'setuptools_scm', 'libclang', 'pyflakes', 'backoff', 'conda', 'pynacl', 'libpng', 'debugpy', 'openpyxl', 'dbus', 'pytz', 'pyqt', 'ldid', 'fonttools', 'lxml', 'nbconvert', 'openjpeg', 'regex', 'msrest', 'terminado', 'ipython_genutils', 'dataclasses', 'pickleshare', 'inotify_simple', 'lzo', 'urllib3', 'vsts-python-api', 'typing_extensions', 'libblas', 'pybind11-abi', 'matplotlib-inline', 'wrapt', 'decorator', 'mistune', 'lua', 'deprecated', 'backports.functools_lru_cache', 'pycparser', 'parso', 'libgfortran5', 'tzdata', 'expat', 'libev', 'future', 'traitlets', 'libmambapy', 'oniguruma', 'ruamel.yaml.clib', 'conda-suggest', 'setuptools_scm_git_archive', 'freetype', 'cryptography', 'flake8']

pkgs/main/noarch         [====================] (00m:00s) No change
pkgs/r/osx-64            [====================] (00m:00s) No change
pkgs/main/osx-64         [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change
conda-forge/noarch       [====================] (00m:02s) Done
conda-forge/osx-64       [====================] (00m:05s) Done

Pinned packages:
  - python 3.8.*


Transaction

  Prefix: /Users/beckermr/miniconda3

  Updating specs:

   - libcurl
   - blinker
   - pexpect
   - gettext
   - libgcrypt
   - jsonschema
   - argon2-cffi
   - pyyaml
   - tomlkit
   - matplotlib-base
   - certifi
   - mamba
   - py-lief
   - nbformat
   - six
   - pygments
   - gitpython
   - keyring
   - liblief
   - pyzmq
   - toml
   - jq
   - qtconsole
   - notebook
   - brotli
   - iniconfig
   - ncurses
   - requests
   - zeromq
   - libmamba
   - libgpg-error
   - jupyter_console
   - cloudpickle
   - pynamodb
   - bleach
   - botocore
   - parameterized
   - send2trash
   - mccabe
   - libuv
   - make
   - rfc3986
   - pyqt-impl
   - lcms2
   - filelock
   - cycler
   - attrs
   - jupyterlab_pygments
   - pkg-config
   - testpath
   - reproc-cpp
   - reproc
   - popt
   - libglib
   - tornado
   - glib-tools
   - backcall
   - cffi
   - pandoc
   - ruamel.yaml.jinja2
   - pyparsing
   - libcxx
   - glob2
   - python-dateutil
   - zlib
   - boa
   - kiwisolver
   - pyopenssl
   - jwcrypto
   - conda-forge-pinning
   - liblapack
   - pysocks
   - coverage
   - libpq
   - uritemplate
   - lerc
   - nest-asyncio
   - more-itertools
   - simplejson
   - appdirs
   - qt
   - smmap
   - pyrsistent
   - libwebp-base
   - importlib_metadata
   - pcre
   - libedit
   - conda-build
   - ptyprocess
   - commonmark
   - pyasn1
   - python-libarchive-c
   - packaging
   - glib
   - numpy
   - jupyter_highlight_selected_word
   - prometheus_client
   - conda-package-handling
   - github3.py
   - pycosat
   - pandocfilters
   - python_abi
   - setproctitle
   - libiconv
   - ipywidgets
   - pyqtwebengine
   - nb_conda_kernels
   - pyjwt
   - munkres
   - backports
   - mysql-common
   - platformdirs
   - olefile
   - cfitsio
   - wheel
   - webencodings
   - xz
   - charset-normalizer
   - chardet
   - jinja2
   - joblib
   - git
   - brotli-bin
   - pip
   - perl
   - appnope
   - setuptools-scm
   - zipp
   - curl
   - license-expression
   - brotlipy
   - colorama
   - gitdb
   - nbclient
   - psutil
   - toolz
   - requests-toolbelt
   - rpm-tools
   - ipython
   - scrypt
   - tk
   - jbig
   - boto3
   - ruamel_yaml
   - jupyter_contrib_nbextensions
   - archspec
   - requests-mock
   - ipykernel
   - libtiff
   - click
   - libgfortran
   - tomli
   - jpeg
   - py
   - pathspec
   - pycrypto
   - sqlite
   - libbrotlidec
   - krb5
   - soupsieve
   - watchgod
   - oauthlib
   - prompt-toolkit
   - hexalattice
   - conda-suggest-conda-forge
   - readme_renderer
   - libcblas
   - widgetsnbextension
   - black
   - docutils
   - libdeflate
   - icu
   - llvm-openmp
   - libffi
   - requests-oauthlib
   - jedi
   - defusedxml
   - typing-extensions
   - uritemplate.py
   - libnghttp2
   - jupyter
   - libdb
   - prompt_toolkit
   - conda-smithy
   - rich
   - libsolv
   - jupyter_latex_envs
   - mysql-libs
   - zstd
   - mock
   - nspr
   - libopenblas
   - file
   - libarchive
   - jupyter_contrib_core
   - pyqtchart
   - frozendict
   - jupyter_nbextensions_configurator
   - tar
   - typed-ast
   - libllvm11
   - m2r
   - importlib_resources
   - yaml
   - jupyter_core
   - entrypoints
   - readline
   - openssl
   - ndg-httpsclient
   - libxml2
   - twine
   - libbrotlicommon
   - anaconda-client
   - loky
   - markupsafe
   - boolean.py
   - pluggy
   - libzlib
   - bzip2
   - libmagic
   - c-ares
   - jupyter_client
   - beautifulsoup4
   - json5
   - lazyasd
   - python
   - cmarkgfm
   - pkginfo
   - ripgrep
   - nodejs
   - s3transfer
   - et_xmlfile
   - pytest-cov
   - argcomplete
   - libxslt
   - unicodedata2
   - pyqt5-sip
   - libbrotlienc
   - ruamel.yaml
   - setuptools
   - libssh2
   - wcwidth
   - yaml-cpp
   - matplotlib
   - jupyterlab_widgets
   - libsodium
   - jdcal
   - clyent
   - pygithub
   - pcre2
   - isodate
   - codecov
   - lz4-c
   - ca-certificates
   - pillow
   - python.app
   - importlib-metadata
   - idna
   - async_generator
   - pytest
   - mypy_extensions
   - tqdm
   - cachetools
   - pycodestyle
   - wurlitzer
   - qtpy
   - nss
   - jmespath
   - setuptools_scm
   - libclang
   - pyflakes
   - backoff
   - conda
   - pynacl
   - libpng
   - debugpy
   - openpyxl
   - dbus
   - pytz
   - pyqt
   - ldid
   - fonttools
   - lxml
   - nbconvert
   - openjpeg
   - regex
   - msrest
   - terminado
   - ipython_genutils
   - dataclasses
   - pickleshare
   - inotify_simple
   - lzo
   - urllib3
   - vsts-python-api
   - typing_extensions
   - libblas
   - pybind11-abi
   - matplotlib-inline
   - wrapt
   - decorator
   - mistune
   - lua
   - deprecated
   - backports.functools_lru_cache
   - pycparser
   - parso
   - libgfortran5
   - tzdata
   - expat
   - libev
   - future
   - traitlets
   - libmambapy
   - oniguruma
   - ruamel.yaml.clib
   - conda-suggest
   - setuptools_scm_git_archive
   - freetype
   - cryptography
   - flake8
   - ca-certificates
   - certifi
   - openssl


  Package                            Version  Build         Channel                  Size
───────────────────────────────────────────────────────────────────────────────────────────
  Upgrade:
───────────────────────────────────────────────────────────────────────────────────────────

  - boto3                            1.20.20  pyhd8ed1ab_0  installed                    
  + boto3                            1.20.21  pyhd8ed1ab_0  conda-forge/noarch      71 KB
  - botocore                         1.23.20  pyhd8ed1ab_0  installed                    
  + botocore                         1.23.21  pyhd8ed1ab_0  conda-forge/noarch       5 MB
  - conda-forge-pinning  2021.12.05.20.40.01  hd8ed1ab_0    installed                    
  + conda-forge-pinning  2021.12.07.13.39.21  hd8ed1ab_0    conda-forge/noarch      18 KB
  - glib                              2.70.1  hcf210ce_0    installed                    
  + glib                              2.70.2  hcf210ce_0    conda-forge/osx-64     425 KB
  - glib-tools                        2.70.1  hcf210ce_0    installed                    
  + glib-tools                        2.70.2  hcf210ce_0    conda-forge/osx-64      93 KB
  - libglib                           2.70.1  hf1fb8c0_0    installed                    
  + libglib                           2.70.2  hf1fb8c0_0    conda-forge/osx-64       3 MB

  Summary:

  Upgrade: 6 packages

  Total download: 9 MB

───────────────────────────────────────────────────────────────────────────────────────────

Finished glib-tools                           (00m:00s)              93 KB    334 KB/s
Finished conda-forge-pinning                  (00m:00s)              18 KB     63 KB/s
Finished glib                                 (00m:00s)             425 KB      1 MB/s
Finished boto3                                (00m:00s)              71 KB    201 KB/s
Finished libglib                              (00m:00s)               3 MB      6 MB/s
Finished botocore                             (00m:00s)               5 MB      8 MB/s
Downloading  [====================================================================================================] (00m:05s)   13.55 MB/s
Extracting   [====================================================================================================] (00m:05s)        6 / 6
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

Notice it reports python as pinned.

Comment (isuruf) [reply conda_smithy/templates/build_steps.sh.tmpl:41]: Can you open a PR in a feedstock with these changes?

Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:41]: Yes!

Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:41]: See here: https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock/pull/321

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 5]
```diff
@@ -37,10 +37,10 @@ BUILD_CMD=build
 CONDA_INSTALL_CMD=conda
 {%- endif %}
 
+${CONDA_INSTALL_CMD} install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
 {%- if build_with_mambabuild %}
-${CONDA_INSTALL_CMD} update --all --yes --quiet -c conda-forge
+${CONDA_INSTALL_CMD} update --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
```
Comment (beckermr) [inline conda_smithy/templates/build_steps.sh.tmpl:42]: This change needs to be pushed to osx too.

**Review by isuruf — APPROVED** [review]

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
Should we pass the update spec flag or something so that this doesn't break later? 

Cc @chenghlee

**Comment by beckermr** [comment]
mamba ignores the flag, so I think we should add it now so that if it ever does something it will work:

```
$ mamba install --update-specs fitsio

Looking for: ['fitsio']

conda-forge/osx-64       Using cache
conda-forge/noarch       Using cache
pkgs/main/noarch         [====================] (00m:00s) No change
pkgs/main/osx-64         [====================] (00m:00s) No change
pkgs/r/osx-64            [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change

Pinned packages:
  - python 3.8.*


Transaction

  Prefix: /Users/beckermr/miniconda3

  Updating specs:

   - fitsio
   - ca-certificates
   - certifi
   - openssl


  Package   Version  Build           Channel                  Size
────────────────────────────────────────────────────────────────────
  Install:
────────────────────────────────────────────────────────────────────

  + fitsio    1.1.5  py38hf771872_1  conda-forge/osx-64     Cached

  Summary:

  Install: 1 packages

  Total download: 0  B

────────────────────────────────────────────────────────────────────

Confirm changes: [Y/n] n
Aborted.
(base) clarence:parsl beckermr$ mamba update --update-specs numpy

Looking for: ['numpy']

conda-forge/osx-64       Using cache
conda-forge/noarch       Using cache
pkgs/main/osx-64         Using cache
pkgs/main/noarch         Using cache
pkgs/r/osx-64            Using cache
pkgs/r/noarch            Using cache

Pinned packages:
  - python 3.8.*


Transaction

  Prefix: /Users/beckermr/miniconda3

  All requested packages already installed

```

File: conda_smithy/templates/build_steps.sh.tmpl [hunk 6]
```diff
@@ -25,15 +27,20 @@ conda-build:
...
+{%- set CONDA_INSTALL_CMD="mamba" %}
+{%- set GET_BOA="" %}
+{%- set BUILD_CMD="build" %}
 {%- endif %}
 
-conda install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
+{{ CONDA_INSTALL_CMD }} install --yes --quiet "{{ remote_ci_setup }}" conda-build pip {{ GET_BOA }}-c conda-forge
+{%- if build_with_mambabuild %}
+{{ CONDA_INSTALL_CMD }} update --yes --quiet "{{ remote_ci_setup }}" conda-build pip {{ GET_BOA }}-c conda-forge
+{%- endif %}
```
Comment (beckermr) [inline conda_smithy/templates/build_steps.sh.tmpl:43]:
```suggestion
{{ CONDA_INSTALL_CMD }} install --update-specs --yes --quiet "{{ remote_ci_setup }}" conda-build pip {{ GET_BOA }}-c conda-forge
{%- if build_with_mambabuild %}
{{ CONDA_INSTALL_CMD }} update --update-specs --yes --quiet "{{ remote_ci_setup }}" conda-build pip {{ GET_BOA }}-c conda-forge
{%- endif %}
```

Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:43]: Sorry to bother you @isuruf, but can you look at this? I think with this commit we should be future-proof and good to merge.

Comment (isuruf) [reply conda_smithy/templates/build_steps.sh.tmpl:43]: Did you try this in a feedstock?

Comment (beckermr) [reply conda_smithy/templates/build_steps.sh.tmpl:43]: Nope but I will go ahead with that.

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 7]
```diff
@@ -9,27 +11,31 @@ MINIFORGE_HOME=${MINIFORGE_HOME:-${HOME}/miniforge3}
...
-
 source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
 echo -e "\n\nInstalling {{ remote_ci_setup }} and conda-build."
-conda install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-}
+{{ CONDA_INSTALL_CMD }} install -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip{{ GET_BOA }}
+{%- if build_with_mambabuild %}
+{{ CONDA_INSTALL_CMD }} update -n base --quiet --yes "{{ remote_ci_setup }}" conda-build pip{{ GET_BOA }}
+{%- endif %}
```
Comment (beckermr) [inline conda_smithy/templates/run_osx_build.sh.tmpl:38]:
```suggestion
{{ CONDA_INSTALL_CMD }} install -n base --update-specs --quiet --yes "{{ remote_ci_setup }}" conda-build pip{{ GET_BOA }}
{%- if build_with_mambabuild %}
{{ CONDA_INSTALL_CMD }} update -n base --update-specs --quiet --yes "{{ remote_ci_setup }}" conda-build pip{{ GET_BOA }}
{%- endif %}
```

**Comment by jakirkham** [comment]
> mamba ignores the flag

Just a question here, are we sure there isn't a non-zero exit code when doing that?

**Comment by beckermr** [comment]
We test it live and the CI builds. So I think its fine.

**Pull request merged**